File: app/src/main/java/jp/gr/java_conf/snake0394/loglook_android/storage/MyShipListFragmentPrefs.java
package jp.gr.java_conf.snake0394.loglook_android.storage;

import java.util.List;
import java.util.Map;

import io.github.kobakei.spot.annotation.Pref;
import io.github.kobakei.spot.annotation.PrefField;
import jp.gr.java_conf.snake0394.loglook_android.storage.converter.LabelListTypeConverter;
import jp.gr.java_conf.snake0394.loglook_android.storage.converter.LabelMapTypeConverter;
import jp.gr.java_conf.snake0394.loglook_android.view.fragment.MyShipListRecyclerViewAdapter;

/**
 * {@link jp.gr.java_conf.snake0394.loglook_android.view.fragment.MyShipListFragment}
 */
@Pref(name = "pref_my_ship_list_fragment")
public class MyShipListFragmentPrefs {

    @PrefField(name = "labelFilter")
    public String labelFilter = "すべて";

    @PrefField(name = "shipTypeFilter")
    public String shipTypeFilter = "すべて";

    @PrefField(name = "sortType")
    public String sortType = "Lv";

    @PrefField(name = "order")
    public String order = "降順";

    @PrefField(name = "toLabelMap", converter = LabelMapTypeConverter.class)
    public Map<Integer, List<MyShipListRecyclerViewAdapter.Label>> toLabelMap;

    @PrefField(name = "labelList", converter = LabelListTypeConverter.class)
    public List<MyShipListRecyclerViewAdapter.Label> labelList;

    public MyShipListFragmentPrefs() {
        //Empty constructor is needed
    }
}

File: app/src/main/java/jp/gr/java_conf/snake0394/loglook_android/storage/MyShipListFragmentPrefs.kt
package jp.gr.java_conf.snake0394.loglook_android.storage

import android.content.Context
import jp.gr.java_conf.snake0394.loglook_android.storage.converter.mapPreference
import jp.gr.java_conf.snake0394.loglook_android.view.fragment.MyShipListRecyclerViewAdapter
import jp.takuji31.koreference.KoreferenceModel
import jp.takuji31.koreference.gson.gsonPreference
import jp.takuji31.koreference.stringPreference

/**
 * [jp.gr.java_conf.snake0394.loglook_android.view.fragment.MyShipListFragment]
 */
class MyShipListFragmentPrefs(context: Context) : KoreferenceModel(context = context, name = "pref_my_ship_list_fragment") {

    var labelFilter by stringPreference("すべて")

    var shipTypeFilter by stringPreference("すべて")

    var sortType by stringPreference("Lv")

    var order by stringPreference("降順")

    var toLabelMap: Map<Int, List<MyShipListRecyclerViewAdapter.Label>> by mapPreference(HashMap<Int, List<MyShipListRecyclerViewAdapter.Label>>())

    var labelList: List<MyShipListRecyclerViewAdapter.Label> by gsonPreference(default =  ArrayList<MyShipListRecyclerViewAdapter.Label>())
}